### PR TITLE
[SG-677]  DNS service verification using DNSClient

### PR DESCRIPTION
**Pull request by gbubemismith**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Create DNS service verification using DNSClient.NET. The service should be able to determine if an organization owns a domain by checking if the TXT record exists on the domain. 

## Code changes

* **src/Core/Services/IDnsResolverService.cs:** Added DNS resolver abstraction which takes in the domain name and txt record as parameters.
* **src/Core/Services/Implementations/DnsResolverService.cs:** Implementation of the `ResolveAsync` method using `DNSClient.Net` to check if the txt record exists on the organization's domain.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: bitwarden_license/src/Commercial.Core/packages.lock.json [hunk 1]
```diff
@@ -2519,41 +2527,42 @@
...
-          "SendGrid": "9.27.0",
-          "Sentry.Serilog": "3.16.0",
-          "Serilog.AspNetCore": "5.0.0",
-          "Serilog.Extensions.Logging": "3.1.0",
-          "Serilog.Extensions.Logging.File": "2.0.0",
-          "Serilog.Sinks.AzureCosmosDB": "2.0.0",
-          "Serilog.Sinks.SyslogMessages": "2.0.6",
-          "Stripe.net": "40.0.0",
-          "YubicoDotNetClient": "1.2.0"
+          "AWSSDK.SQS": "[3.7.2.47, )",
```
Comment (differsthecat) [inline bitwarden_license/src/Commercial.Core/packages.lock.json:2530]: What do these changes do, adding the `[ ,)` around the versions?

Comment (gbubemismith) [reply bitwarden_license/src/Commercial.Core/packages.lock.json:2530]: I think that came from a merge from `master`

Comment (differsthecat) [reply bitwarden_license/src/Commercial.Core/packages.lock.json:2530]: Got it, can we merge master into the feature branch then? That will clean up this diff a lot and then we can be sure!

Comment (differsthecat) [reply bitwarden_license/src/Commercial.Core/packages.lock.json:2530]: I can do that if you'd like

Comment (gbubemismith) [reply bitwarden_license/src/Commercial.Core/packages.lock.json:2530]: No problems I would do that

**Review by differsthecat — APPROVED** [review]
Look good, awesome Gbubemi!

**Pull request merged**